daemon/daemon.go

```diff
@@ -101,6 +101,24 @@ func NewDaemon(cfg *config.Config) *Daemon {
 		return nil
 	}
 
+	// get containerd version to make sure containerd has started
+	_, err = ctrdClient.Version(context.Background())
+	if err != nil {
+		logrus.Errorf("failed to get containerd version: %v", err)
+		return nil
+	}
+
+	if cfg.Snapshotter != "" {
+		ctrd.SetSnapshotterName(cfg.Snapshotter)
+	}
+
+	if err = checkSnapshotter(ctrd.CurrentSnapshotterName(), filepath.Join(cfg.HomeDir, "containerd/root")); err != nil {
```

Comment:
I don't think we should check the dir name to get the snapshot. The containerd provides plugin API to get the detail of loading plugin. by ctr command, we can get the detail like 

```
➜  containerd git:(773c489) sudo ctr plugin
TYPE                            ID             PLATFORMS      STATUS
io.containerd.content.v1        content        -              ok
io.containerd.snapshotter.v1    btrfs          linux/amd64    error
io.containerd.snapshotter.v1    overlayfs      linux/amd64    ok
io.containerd.metadata.v1       bolt           -              ok
io.containerd.differ.v1         walking        linux/amd64    ok
io.containerd.gc.v1             scheduler      -              ok
io.containerd.grpc.v1           containers     -              ok
io.containerd.grpc.v1           content        -              ok
io.containerd.grpc.v1           diff           -              ok
io.containerd.grpc.v1           events         -              ok
io.containerd.grpc.v1           healthcheck    -              ok
io.containerd.grpc.v1           images         -              ok
io.containerd.grpc.v1           leases         -              ok
io.containerd.grpc.v1           namespaces     -              ok
io.containerd.grpc.v1           snapshots      -              ok
io.containerd.monitor.v1        cgroups        linux/amd64    ok
io.containerd.runtime.v1        linux          linux/amd64    ok
io.containerd.grpc.v1           tasks          -              ok
io.containerd.grpc.v1           version        -              ok
```

Reply:
Got it!